src/Microsoft.Azure.WebJobs.Host/Hosting/DrainModeManager.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.Logging;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    internal class DrainModeManager : IDrainModeManager
+    {
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly ILogger _logger;
+
+        public DrainModeManager(ILoggerFactory loggerFactory = null)
+        {
+            _loggerFactory = loggerFactory;
+            _logger = _loggerFactory.CreateLogger<DrainModeManager>();
+        }
+
+        public bool DrainModeEnabled { get; private set; } = false;
```

Comment:
nit -- can you make this `IsDrainModeEnabled`?

Reply:
done